autosklearn/evaluation/__init__.py

```diff
@@ -85,23 +98,30 @@ def fit_predict_try_except_decorator(
         queue.close()
 
 
-def get_cost_of_crash(metric: Scorer) -> float:
-
-    # The metric must always be defined to extract optimum/worst
-    if not isinstance(metric, Scorer):
-        raise ValueError("The metric must be stricly be an instance of Scorer")
-
-    # Autosklearn optimizes the err. This function translates
-    # worst_possible_result to be a minimization problem.
-    # For metrics like accuracy that are bounded to [0,1]
-    # metric.optimum==1 is the worst cost.
-    # A simple guide is to use greater_is_better embedded as sign
-    if metric._sign < 0:
-        worst_possible_result = metric._worst_possible_result
-    else:
-        worst_possible_result = metric._optimum - metric._worst_possible_result
+def get_cost_of_crash(metrics: Sequence[Scorer]) -> List[float] | float:
+    """Return the cost of crash.
+
+    Return value can be either a list (multi-objective optimization) or a
+    raw float (single objective) because SMAC assumes different types in the
+    two different cases.
+    """
+    costs = []
+    for metric in metrics:
+        if not isinstance(metric, Scorer):
```

Comment:
This should not be possible due to the signature

Reply:
That's correct, but this check has been here before, so I'd rather keep it.